### ISSUE TITLE
A file deleted from all nodes may exist in the "out of sync" list

### ISSUE THREAD

**Issue by TianyiChen** [syncthing/syncthing#4627]
When a file is deleted from the slave first, and then from the read-only master. It disappeared and later existed again in the "out of sync" list on the master. I feel it should be excluded permanently in the list.

My steps to reproduce:
- Create 4 files 0, 1, 2, 3 on the master node. Set the folder in the master to be "send only"
- Do an initial synchronization
- Delete 2, 3 on the slave
- Rescan on the slave and synchronize
- Delete 1, 2 on the master
- Pause and resume on the master (optional)
- The "out of sync" list shows only 3
- Later (I tried 3 times)
  - In 2 attempts, the final list showed after a pause and a resume
  - In 1 attempt, I paused and resumed several times before seeing the count of "out of sync" list to be 2
- Now we have 0, 3 on the master and 0 on the slave
- Eventually the "out of sync" list shows 2 and 3. Since 2 is already deleted from the master and the slave, I think it shouldn't appear here

I tried to restart Syncthing or the operation system, but the problem persisted

Screenshots from the master:

<img width="514" alt="untitled" src="https://user-images.githubusercontent.com/10559772/34458213-28c48446-ed8f-11e7-8ae7-1321c1b6ae93.png">

<img width="353" alt="untitled2" src="https://user-images.githubusercontent.com/10559772/34458304-53922036-ed92-11e7-9a10-4fa42b164876.png">

### Version Information

Syncthing Version: v0.14.42
OS Version: 

Master: Windows7 (64-bit)  Slave: Android 7.1
I also encountered a similar issue with Master: Android 4.4    Slave: Android 7.1

**Comment by AudriusButkevicius** [comment]
Sorry, what's the issue, you deleted 2, 3 on slave, and master refuses to accept that (as it's master), hence becomes out of sync.

Anyways, support is handled on the forums.

**Comment by TianyiChen** [comment]
I know that the state of out of sync is expected in this case. 2 is deleted from the slave and the master so it no longer exists, should it still exist in the out of sync list?

**Comment by AudriusButkevicius** [comment]
Logically it doesn't make sense, yet technically because even the state is the same, the action is initiated by different devices, hence it's out of sync.
I'll leave for @calmh to answer.

**Comment by calmh** [comment]
No, it’s a bug. What should logically happen I think is

1. Things are in sync
2. File deleted in normal folder, other device with send only folder is out of sync. “Override” will resurrect 
3. File deleted in send only folder. Things are in sync. 

It’s similar to the other issue when a new device with identical contents coming online will make a send only device fully out of sync. 

We should resolve the differences as long as it doesn’t result in differences on disk for the send only folder.

**Comment by tr37ion** [comment]
For me, is not fixed with **v0.14.49 Linux (64 bit)**. Regression?

**Comment by imsodin** [comment]
For me the steps above result in the expected up-to-date state. Please post on forum.syncthing.net until we identified and reproduced the problem.

**Comment by tr37ion** [comment]
From what I remember, I did try ...

 1. Copying a large (5GB) folder with many small files into my sync folder of host A.
 2. Then Syncthing picked it up and started syncing.
 3. After quite some time (still in sync process) I had to leave and I decided to delete the folder
 4. Therefore, I did delete the folder on A - with the hope Syncthing would instantly stop syncing and deelte everything on target host B
 5. Well, This did a lot longer than expected - I had to leave - So, I just deleted the folder on B, too. (half-synced)

 .. and I think a t that point I did mess up something. As this resulted in Syncthing keeping the missing files as "unsynced". Which is quite obvious, as the sopurce wasn't present anymore and the marked files for deletion weren't there either.
